CMake/setup.cmake

```diff
@@ -117,17 +117,22 @@ if(APPLE)
   set(HAVE_VSNPRINTF 1)
   set(HAVE_SCANDIR 1)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated")
+  if (NOT DEFINED ${CMAKE_OSX_DEPLOYMENT_TARGET})
+    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.8)
+  endif()
```

Comment:
Setting this is a normal CMake variable is probably not a good idea since the real variable is stored in the cache, and doing it like above will "shadow" the cache variable, i.e. the user doesn't have much control and can't "see" the contents and can't effectively change it.
I suggest to change the assignment to:
```
  set(CMAKE_OSX_DEPLOYMENT_TARGET 10.8 CACHE STRING "<comment>")
```
I'm not sure about the comment though. See more in my comments in "Conversation". Needs testing...